### PR TITLE
Ports typing indicators using your bubble icon from tgstation

### LINKED ISSUE #4895: "Lawyerspeak" emotes are gone
## Round ID:
Every round

## Testmerges:

Could've been caused by the "recent" runechat/overhead chat update to include emotes. Not 100% sure though.

## Reproduction:

Normally when you play as a lawyer and attach your attorney's badge to your jumpsuit there should be some nice emotes appearing next to your head when you speak. For example a big red exclamation mark, or question mark for questions. It gives that ace attorney vibe. Well all of this is just gone. It doesn't work anymore. It's been like this for a long while too. I am not a 100% sure but i think it stopped working when the overhead chat got updated.

### PR DESCRIPTION
**Pull request by Miliviu**
## About The Pull Request

Ports https://github.com/tgstation/tgstation/pull/67872
Now the Typing Indicator uses your bubble icon 
Closes #4895 

## Why It's Good For The Game

We get back the cool Lawyer and other bubble indicators and I mean @Fresh341 really wanted the badge to work again!

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Video showing the new Lawyer typing indicator and the normal one with the Psychozia for no reason at all!

https://github.com/BeeStation/BeeStation-Hornet/assets/45698448/51dc0e8f-6371-4d6e-ab75-995ddb1ff902

</details>

## Changelog
:cl:Crumpaloo, jlsnow301, JohnFulpWillard, Miliviu
add: Typing indicators now use your speech bubble type, rather than all being the same, so Lawyers have that cool speech bubble again.
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
I'm not sure that this is a good idea. What if the player's job or species is obscured in some way and they start typing, meta-exposing them?

**Comment by Miliviu** [comment]
> I'm not sure that this is a good idea. What if the player's job or species is obscured in some way and they start typing, meta-exposing them?

I don't know, it's something that you can already can see if you disable Runechat (bubble_icons), so, If it wasn't an issue before I'd be weird if it was an issue today, right?

**Comment by Miliviu** [comment]
> I'm not sure that this is a good idea. What if the player's job or species is obscured in some way and they start typing, meta-exposing them?

I don't know, it's something that you can already can see if you disable Runechat (bubble_icons), so, If it wasn't an issue before I'd be weird if it was an issue today, right?

**Comment by itsmeow** [comment]
Because nobody disables them, and they haven't been around for everyone since the server's LRP days.

**Comment by JixS4v** [comment]
Isn't this just tied to the badge? People can still remove it

**Comment by Miliviu** [comment]
It's not just the badge there are some more, let me check if any of them is problematic!

**Comment by JixS4v** [comment]
You can still tell someone's species by just checking the flavourtext like "blorbles", "meows", "hisses"

**Comment by Miliviu** [comment]
1. We have the default bubbles, no issue with them
2. Then Alien and Alien Royale, there is no way or need to obscure that afaik
3. Machine is used by silicons, I don't know how a borg or a Silicon could try to obscure their state as borgs 
4.  Syndiborgs, this one kinda concerns me, Is there a way to obscure your Syndicate Status as a borg?
5. The rest of em are for like, Clockcult Scarabs, swarmers...
6. Finally the lawyer, you can just remove the badge and tadaa

SO, if a Syndie Cyborg uses chameleon their buble icon changes to Standar Cyborg

**Comment by Miliviu** [comment]
https://github.com/BeeStation/BeeStation-Hornet/blob/87794d6eb4e652b48ac0f627b86efc56b04f60df/code/modules/antagonists/nukeop/equipment/borgchameleon.dm#L85-L93

**Comment by Miliviu** [comment]
So, honestly, I think there isn't a case where this can "reveal" your Species, Traitor Status, etc

**Comment by Tsar-Salat** [comment]
There are enough tells to see someone's species, especially ones more obvious than speech, that I don't think it'll be an issue.

**Review by itsmeow — CHANGES_REQUESTED** [review]
This won't work with blindness

**Comment by Miliviu** [comment]
It'll not?

**Comment by itsmeow** [comment]
No, looking at the code you haven't set the planes and layers properly or updated the blindness indicator at all

**Comment by Miliviu** [comment]
I'll draft this and take a peak asap!

**Comment by itsmeow** [comment]
More specifically, the old indicators have these planes/layers:
image

File: code/modules/tgui_input/say_modal/typing.dm [hunk 1]
```diff
@@ -76,30 +71,30 @@ GLOBAL_DATUM_INIT(typing_indicator, /mutable_appearance, mutable_appearance('ico
...
+	active_thinking_indicator = FALSE
 
 /mob/living/create_typing_indicator(override = FALSE)
-	if(typing_indicator || ((thinking_indicator || !thinking_IC) && !override) || stat != CONSCIOUS)
+	if(active_typing_indicator || ((active_thinking_indicator || !thinking_IC) && !override) || stat != CONSCIOUS)
 		return FALSE
-	add_overlay(GLOB.typing_indicator)
-	add_overlay(GLOB.blind_typing_indicator)
-	typing_indicator = TRUE
+	active_blind_typing_indicator = mutable_appearance('icons/mob/talk.dmi', "[bubble_icon]0", (-TYPING_LAYER), BLIND_FEATURE_PLANE, appearance_flags = KEEP_TOGETHER)
```
Comment (Miliviu) [inline code/modules/tgui_input/say_modal/typing.dm:88]: This doesn't do it? @itsmeow

**Review by PowerfulBacon — APPROVED** [review]

**Review by Rukofamicom — APPROVED** [review]

File: code/modules/tgui_input/say_modal/typing.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/tgui_input/say_modal/typing.dm:88]: Did you.. test it?

Comment (Miliviu) [reply code/modules/tgui_input/say_modal/typing.dm:88]: I only tested it without blind people oops, aaand you were right, it breaks it, I'll fix it asap

**Comment by Miliviu** [comment]
Errr, I lost interest on this, I need to figure it out with blind code, whenever I feel like doing it I'll reopen the PR!

**Pull request closed without merging**